apps/cuda_mat_mul/runner.cpp

```diff
@@ -1,3 +1,8 @@
+// We always want asserts
+#ifndef NDEBUG
+#undef NDEBUG
+#endif
```

Comment:
This is definitely wrong. Should be `#ifdef NDEBUG`

Reply:
Could have sworn I just fixed that. Must have forgotten to push.

Reply:
Ah, that's not part of this PR. Please see #5034, on which this was based. Once that's done I'll rebase this on top of the result.

Reply:
Rebased on master instead. This change has been removed from this PR

Reply:
Great, thanks!